tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/AWS/LambdaCommon.cs

```diff
@@ -54,64 +53,68 @@ internal static CallTargetState StartInvocationOneParameter<TArg>(ILambdaExtensi
         internal static CallTargetReturn<TReturn> EndInvocationSync<TReturn>(TReturn returnValue, Exception exception, Scope scope, ILambdaExtensionRequest requestBuilder)
         {
             var json = SerializeObject(returnValue);
-            scope?.Dispose();
             Flush();
-            NotifyExtensionEnd(requestBuilder, exception != null, json);
+            NotifyExtensionEnd(requestBuilder, scope, exception != null, json);
+            scope?.Dispose();
             return new CallTargetReturn<TReturn>(returnValue);
         }
 
         internal static TReturn EndInvocationAsync<TReturn>(TReturn returnValue, Exception exception, Scope scope, ILambdaExtensionRequest requestBuilder)
         {
             var json = SerializeObject(returnValue);
-            scope?.Dispose();
             Flush();
-            NotifyExtensionEnd(requestBuilder, exception != null, json);
+            NotifyExtensionEnd(requestBuilder, scope, exception != null, json);
+            scope?.Dispose();
             return returnValue;
         }
 
         internal static CallTargetReturn EndInvocationVoid(Exception exception, Scope scope, ILambdaExtensionRequest requestBuilder)
         {
             scope?.Dispose();
             Flush();
-            NotifyExtensionEnd(requestBuilder, exception != null);
+            NotifyExtensionEnd(requestBuilder, scope, exception != null);
             return CallTargetReturn.GetDefault();
         }
 
-        internal static Scope CreatePlaceholderScope(Tracer tracer, ILambdaExtensionRequest requestBuilder)
+        internal static Scope CreatePlaceholderScope(Tracer tracer, string traceId, string samplingPriority)
         {
-            Scope scope = null;
-            try
+            Span span = null;
+            if (traceId == null || samplingPriority == null)
             {
-                var request = requestBuilder.GetTraceContextRequest();
-                using (var response = (HttpWebResponse)request.GetResponse())
-                {
-                    var traceId = response.Headers.Get(HttpHeaderNames.TraceId);
-                    // need to set the exact same spanId so nested spans (auto-instrumentation or manual) will have the correct parent-id
-                    var spanId = response.Headers.Get(HttpHeaderNames.SpanId);
-                    Serverless.Debug($"received traceId = {traceId} and spanId = {spanId}");
-                    var span = tracer.StartSpan(PlaceholderOperationName, null, serviceName: PlaceholderServiceName, traceId: Convert.ToUInt64(traceId), spanId: Convert.ToUInt64(spanId), addToTraceContext: false);
-                    scope = tracer.TracerManager.ScopeManager.Activate(span, false);
-                }
+                Serverless.Debug("samplingPriority and/or traceId not found");
+                span = tracer.StartSpan(PlaceholderOperationName, null, serviceName: PlaceholderServiceName, addToTraceContext: false);
```

Comment:
tiny nit: prefer providing named args for null params
```suggestion
                span = tracer.StartSpan(PlaceholderOperationName, tags: null, serviceName: PlaceholderServiceName, addToTraceContext: false);
```

Reply:
✅